src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -0,0 +1,1218 @@
+import htsjdk.samtools.util.Locatable;
+import htsjdk.tribble.Feature;
+import htsjdk.tribble.annotation.Strand;
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.ReferenceDataSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.funcotator.*;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.codecs.GENCODE.*;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+
+import java.io.File;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * A factory to create {@link GencodeFuncotation}s.
+ * Created by jonn on 8/30/17.
+ */
+public class GencodeFuncotationFactory extends DataSourceFuncotationFactory {
+
+    //==================================================================================================================
+
+    /**
+     * The window around splice sites to mark variants as {@link org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotation.VariantClassification#SPLICE_SITE}.
+     */
+    final static private int spliceSiteVariantWindowBases = 2;
+
+    //==================================================================================================================
+
+    /**
+     * ReferenceSequenceFile for the transcript reference file.
+     */
+    private final ReferenceDataSource transcriptFastaReferenceDataSource;
+
+    /**
+     * Map between transcript IDs and the IDs from the FASTA file to look up the transcript.
+     * This is necessary because of the way the FASTA file contigs are named.
+     */
+    private final Map<String, MappedTranscriptIdInfo> transcriptIdMap;
+
+    //==================================================================================================================
+
+    public GencodeFuncotationFactory(final File gencodeTranscriptFastaFile) {
+        transcriptFastaReferenceDataSource = ReferenceDataSource.of(gencodeTranscriptFastaFile);
+
+        transcriptIdMap = createTranscriptIdMap(transcriptFastaReferenceDataSource);
+    }
+
+    //==================================================================================================================
+
+    @Override
+    public void close() {
+        transcriptFastaReferenceDataSource.close();
+    }
+
+    @Override
+    public List<String> getSupportedFuncotationFields() {
+        return GencodeFuncotation.getSerializedFieldNames();
+    }
+
+    @Override
+    public List<Funcotation> createFuncotations(final VariantContext variant, final ReferenceContext referenceContext, final List<Feature> featureList) {
```

Comment:
Needs javadocs.

Reply:
I added in a brief Javadoc.  The method is primarily commented in its parent class `DataSourceFuncotationFactory`.